Xamarin.Essentials/DeviceInfo/DeviceInfo.android.cs

```diff
@@ -22,6 +23,14 @@ static string GetDeviceName()
             return name;
         }
 
+        static string[] GetArchitectures()
+        {
+            if (Essentials.Platform.HasApiLevel(BuildVersionCodes.Lollipop))
+                return Build.SupportedAbis.ToArray();
```

Comment:
This property returns "Supported ABIs", not the architecture the app runs as. That's different from all the other implementations. If it returned the actual executing ABI, you wouldn't even have to make this an array (It's really odd to me it's an array of architectures as a process can only run as one thing).

Reply:
@dotMorten thanks for your review. You're right, ABI is different from CPU architecture (CA). In my research, I found this method that returns the CA
`Java.Lang.JavaSystem.GetProperty("os.arch"));`
but for some devices, this method doesn't return the correct CA, for example, the [Samsung Galaxy S5 Neo (reference](https://stackoverflow.com/a/49740834)). 

So, maybe, the best way to know the CA is using the `Build.SupportedAbis`... I didn't do a verification, as in the example above, because some times returns more than one value (e.g. Motorola One).

Reply:
I'm a little confused by this. So instead of some devices returning the wrong thing (that's Android for ya), you're going to make almost all devices return a set of things, where all but one are wrong? The intent (I assume) is to get the architecture the app is running under - So a few Android devices are broken in this respects, but that is leading to a design choice that affects all platforms by returning an array instead, and making it useless on Android?

Reply:
One more thing to clarify: Are you looking at returning the architecture of your app, or the architecture of the device?
(ie you can run x86 apps on x64 PCs, or Armv7 on Armv8, so which one do you want to return?)

Reply:
My English is not so good, so sometimes I don't express my self well. So let me rephrase. 
To me returning the supported ABI give, to the developer, a safer result. I mean, Android platform is very fragmented, every day a new device comes out and we don't know how many devices might have the same issue as the Nexus mentioned above.

> One more thing to clarify: Are you looking at returning the architecture of your app, or the architecture of the device?
I assume we are looking for the architecture that the device supports.

If I'm not very clear, please let me know.

Reply:
And, gently asking. When I returning arm64-v8a, this is not mean that the **device architecture** is arm 64bits?

Reply:
I think that SupportedABIs is incorrect then and it should be the device architecture.